### PR TITLE
Update Carpentries Gitpod environment

### PR DESCRIPTION
**Pull request by mahesh-panchal**
Updates the Gitpod environment for Carpentries training.

## TODO
- [x] Install tools for exercises. Which tools need to be installed?
- [x] Which pages should include a Launch in Gitpod button?
- [x] Where is the description on how to use Gitpod?
- [x] Are any additional extensions needed?
- [x] What should the checkout location be?
- [x] What branch should be checked out?

### CODE REVIEW

**Comment by ggrimes** [comment]
Install tools for exercises. Which tools need to be installed?

The tools in the environment.yml file

https://github.com/carpentries-incubator/workflows-nextflow/blob/main/episodes/data/environment.yml

**Comment by mahesh-panchal** [comment]
Do you want those pinned versions for a reason or do you just want the tools installed? For example, Nextflow is pinned, but presumably you want the latest version right?

**Comment by ggrimes** [comment]
You are right , for nextflow, python and nf-core tools non-pinned and for the specific version of the software below.

  - salmon=1.5
  - fastqc=0.11
  - multiqc=1.10

**Comment by mahesh-panchal** [comment]
Those three tools have been added to the environment.

Do we have a specific branch that should be checked out and where should we include launch buttons?

**Comment by ggrimes** [comment]
I don't have a specific branch in mind, the site is rendered from the main branch, if there any reason not to use this?

**Comment by ggrimes** [comment]
Is it possible as part of gitpod environment set up to create this directory and move into it or is this not advised?

```
tasks:
  - init: |
      mkdir -p nf-training  # Create the directory
      cd nf-training        # Change into the directory
```

**Comment by mahesh-panchal** [comment]
Yes, it's possible. 
My preference though would be to create a clean environment free from distracting files. 
But let's start with that. Better something than nothing.

File: .gitpod.yml [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
       sudo gem install bundler
-
+  # Update Nextflow if image version is behind
+  # Install tools for exercises
+  - name: Update tools for exercises
+    command: |
+      conda install --yes --update-all \
+        salmon=1.5 \
+        fastqc=0.11 \
+        multiqc=1.10 \
```
Comment (ggrimes) [inline .gitpod.yml:21]: Can you please the version for multiqc , as I tried the gitpod environment and got an error with this version of multiqc but it worked when I updated it. Thanks

Comment (mahesh-panchal) [reply .gitpod.yml]: It's unpinned now.

**Comment by mahesh-panchal** [comment]
Are there any particular extensions required for vscode that you would like.
Live-server is there so users can view the multiqc reports. Directly clicking on a MQC report from the side will open up the html, but one can right click it and view with the live server instead to view the rendered version.

**Comment by ggrimes** [comment]
> ticular extensions required for vscode that you would like?
> Live-server is there so users can view the multiqc reports. Directly clicking on a MQC report from the side will open up the html, but one can right click it and view with the live server instead to view the rendered version.

I don't have any any particular extensions in mind as I am  able to view rendered html using the environment you have already setup. Would be great if you could put any useful extensions in. Also how does the live server work can it render and display the course website within the gitpod environment?

**Comment by mahesh-panchal** [comment]
I've already added a task to open the course website in a panel. There's a `gp preview` command to do this.

The live server works by right-clicking on a html file and selecting open with live-server.

**Pull request merged**